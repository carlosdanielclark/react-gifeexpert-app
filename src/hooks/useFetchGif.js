import { useState, useEffect } from "react";
import { getImgGif } from "../helpers/getGif";
import "animate.css";

export const useFetchGifs = (category)=>{
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  //Condicion de la peticion
  useEffect(() => {
    getImgGif(category)
    .then((image) => {

      setState({
        data: image,
        loading: false,
      });
      
    });
  }, [category]);


  return state; //{data: [], loading: true}
}