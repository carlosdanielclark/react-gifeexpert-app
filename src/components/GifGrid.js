import React from 'react';
import '../stylesheet/GifGrid.css'
import  GifGridItem from './GifGridItem.js';
import { useFetchGifs } from '../hooks/useFetchGif.js'; 
import { PropTypes } from 'prop-types';

const GifGrid = ({ category }) => {

  const { data:image, loading }=useFetchGifs( category );

  return (
    <> 
      <h3 className='animate_animated animate_fadeIn'>{ category }</h3>
      
      {loading && <p className="animate_animated animate_flash">Loading</p>}

      <div className='GifGrid_container'>
        {image.map((img) => {
          return <GifGridItem key={img.id} title={img.title} url={img.url} />;
        })}
      </div>
    </>
  );
}

export default GifGrid;

GifGrid.propTypes={
  category: PropTypes.string.isRequired
} 