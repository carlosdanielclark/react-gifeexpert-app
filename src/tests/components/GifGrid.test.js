import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'; 
import  GifGrid  from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGif';

jest.mock('../../hooks/useFetchGif'); //Create component

describe('Testing component "GifGrid.js"', ()=>{
  //(1) Inicializacion
  const category = "venom"; 

  //(2) Estimulo
  let wrapper;

  test('Take Snapshot from "GifGrid"', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    wrapper = shallow( <GifGrid 
      category = { category }
    /> ); 

    //(3) Observacion
    expect( wrapper ).toMatchSnapshot();

  });

  test('Must show items when load the images with "useFetchGifs"', () => {
    const gifs = [{
      id: 'abc123',
      url: 'https:/localhost:3000/anythings.jpg',
      title: 'AnyThings'
    },
    {
      id: 'abc345',
      url: 'https:/localhost:3500/anythings.jpg',
      title: 'AnyThings'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    wrapper = shallow( <GifGrid 
      category = { category }
    /> ); 

    //(3) Observacion
    expect( wrapper ).toMatchSnapshot();
    // 1. The paragraph should not exist 
    expect( wrapper.find('p').exists() ).toBe( false );
    // 2. GifGridItem should exist 
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });

});