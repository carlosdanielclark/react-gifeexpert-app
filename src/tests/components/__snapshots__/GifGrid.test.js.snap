// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component "GifGrid.js" Must show items when load the images with "useFetchGifs" 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    venom
  </h3>
  <div
    className="GifGrid_container"
  >
    <GifGridItem
      key="abc123"
      title="AnyThings"
      url="https:/localhost:3000/anythings.jpg"
    />
    <GifGridItem
      key="abc345"
      title="AnyThings"
      url="https:/localhost:3500/anythings.jpg"
    />
  </div>
</Fragment>
`;

exports[`Testing component "GifGrid.js" Take Snapshot from "GifGrid" 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    venom
  </h3>
  <p
    className="animate_animated animate_flash"
  >
    Loading
  </p>
  <div
    className="GifGrid_container"
  />
</Fragment>
`;
